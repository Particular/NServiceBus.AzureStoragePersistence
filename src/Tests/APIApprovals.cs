using NServiceBus;
using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void ApproveAzureStoragePersistence()
    {
        var publicApi = typeof(AzureStoragePersistence).Assembly.GeneratePublicApi(new ApiGeneratorOptions
        {
            ExcludeAttributes = new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" }
        });
        Approver.Verify(publicApi);
    }
}