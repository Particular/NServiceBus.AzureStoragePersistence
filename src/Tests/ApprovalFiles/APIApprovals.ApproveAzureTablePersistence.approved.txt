[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"Persistence.CosmosDB.Tests, PublicKey=00240000048000009400000006020000002400005253413100040000010001006dab16cbf58f948e25e061c40486817b00722158f0a2787d8cd3f503daf44978f96edac144e917e59ec98999e011ddde52c53938a6cb5a87a213aa2b5dfe2d45e9035ca40c63caf36c8692e06bbc3eb717070e8ca5813ef695153429dcc4fd7db4c790c724e959b37d9697f7dc22369a20ae67f4785b424edbc63c732dd9eecf")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"Persistence.StorageTable.Tests, PublicKey=00240000048000009400000006020000002400005253413100040000010001006dab16cbf58f948e25e061c40486817b00722158f0a2787d8cd3f503daf44978f96edac144e917e59ec98999e011ddde52c53938a6cb5a87a213aa2b5dfe2d45e9035ca40c63caf36c8692e06bbc3eb717070e8ca5813ef695153429dcc4fd7db4c790c724e959b37d9697f7dc22369a20ae67f4785b424edbc63c732dd9eecf")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"Tests, PublicKey=00240000048000009400000006020000002400005253413100040000010001006dab16cbf58f948e25e061c40486817b00722158f0a2787d8cd3f503daf44978f96edac144e917e59ec98999e011ddde52c53938a6cb5a87a213aa2b5dfe2d45e9035ca40c63caf36c8692e06bbc3eb717070e8ca5813ef695153429dcc4fd7db4c790c724e959b37d9697f7dc22369a20ae67f4785b424edbc63c732dd9eecf")]
namespace NServiceBus
{
    [System.Obsolete("The persistence has been renamed from AzureStorage to AzureTable. Use `AzureTable" +
        "Persistence` instead. Will be removed in version 4.0.0.", true)]
    public class AzureStoragePersistence : NServiceBus.Persistence.PersistenceDefinition
    {
        public AzureStoragePersistence() { }
    }
    [System.Obsolete("The saga persistence feature is no longer exposed as a public type. Will be remov" +
        "ed in version 4.0.0.", true)]
    public class AzureStorageSagaPersistence
    {
        public AzureStorageSagaPersistence() { }
    }
    [System.Obsolete("The subscription persistence feature is no longer exposed as a public type. Will " +
        "be removed in version 4.0.0.", true)]
    public class AzureStorageSubscriptionPersistence
    {
        public AzureStorageSubscriptionPersistence() { }
    }
    [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
        ". Will be removed in version 4.0.0.", true)]
    public class AzureStorageTimeoutPersistence
    {
        public AzureStorageTimeoutPersistence() { }
    }
    public sealed class AzureTablePersistence : NServiceBus.Persistence.PersistenceDefinition { }
    public static class ConfigureAzureSagaStorage
    {
        [System.Obsolete(@"The migration mode that supports looking up correlated sagas by secondary indexes is by default enabled and assumes no full table scan is required. In order to opt-in for a table scan for sagas stored with version 1.4 or earlier use `AllowSecondaryKeyLookupToFallbackToFullTableScan`. The member currently throws a NotImplementedException. Will be removed in version 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> AssumeSecondaryIndicesExist(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> config) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> ConnectionString(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> config, string connectionString) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> CreateSchema(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> config, bool createSchema) { }
        public static NServiceBus.MigrationSettings Migration(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> config) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> UseCloudTableClient(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Sagas> config, Microsoft.Azure.Cosmos.Table.CloudTableClient client) { }
    }
    public static class ConfigureAzureStorage
    {
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence> ConnectionString(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence> config, string connectionString) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence> DefaultTable(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence> persistenceExtensions, string tableName) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence> UseCloudTableClient(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence> config, Microsoft.Azure.Cosmos.Table.CloudTableClient client) { }
    }
    public static class ConfigureAzureSubscriptionStorage
    {
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> CacheFor(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> config, System.TimeSpan timeSpan) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> ConnectionString(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> config, string connectionString) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> CreateSchema(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> config, bool createSchema) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> TableName(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> config, string tableName) { }
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> UseCloudTableClient(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Subscriptions> config, Microsoft.Azure.Cosmos.Table.CloudTableClient client) { }
    }
    public static class ConfigureAzureTimeoutStorage
    {
        [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
            ". The member currently throws a NotImplementedException. Will be removed in vers" +
            "ion 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> CatchUpInterval(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> config, int catchUpInterval) { }
        [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
            ". The member currently throws a NotImplementedException. Will be removed in vers" +
            "ion 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> ConnectionString(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> config, string connectionString) { }
        [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
            ". The member currently throws a NotImplementedException. Will be removed in vers" +
            "ion 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> CreateSchema(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> config, bool createSchema) { }
        [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
            ". The member currently throws a NotImplementedException. Will be removed in vers" +
            "ion 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> PartitionKeyScope(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> config, string partitionKeyScope) { }
        [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
            ". The member currently throws a NotImplementedException. Will be removed in vers" +
            "ion 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> TimeoutDataTableName(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> config, string tableName) { }
        [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
            ". The member currently throws a NotImplementedException. Will be removed in vers" +
            "ion 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> TimeoutManagerDataTableName(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> config, string tableName) { }
        [System.Obsolete("Azure transports support timeouts natively and do not require timeout persistence" +
            ". The member currently throws a NotImplementedException. Will be removed in vers" +
            "ion 4.0.0.", true)]
        public static NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> TimeoutStateContainerName(this NServiceBus.PersistenceExtensions<NServiceBus.AzureTablePersistence, NServiceBus.StorageType.Timeouts> config, string blobName) { }
    }
    public class MigrationSettings : NServiceBus.Configuration.AdvancedExtensibility.ExposeSettings
    {
        public void AllowSecondaryKeyLookupToFallbackToFullTableScan() { }
        public void AssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey() { }
        public void DisableSecondaryKeyLookupForSagasCorrelatedByProperties() { }
    }
    public readonly struct TableEntityPartitionKey : System.IEquatable<NServiceBus.TableEntityPartitionKey>
    {
        public TableEntityPartitionKey(string partitionKey) { }
        public string PartitionKey { get; }
        public bool Equals(NServiceBus.TableEntityPartitionKey other) { }
        public override bool Equals(object obj) { }
        public override int GetHashCode() { }
        public static bool operator !=(NServiceBus.TableEntityPartitionKey left, NServiceBus.TableEntityPartitionKey right) { }
        public static bool operator ==(NServiceBus.TableEntityPartitionKey left, NServiceBus.TableEntityPartitionKey right) { }
    }
    public readonly struct TableInformation
    {
        public TableInformation(string tableName) { }
        public string TableName { get; }
    }
}
namespace NServiceBus.Persistence.AzureTable
{
    public class DuplicatedSagaFoundException : System.Exception
    {
        public DuplicatedSagaFoundException(System.Type sagaType, string propertyName, params System.Guid[] identifiers) { }
        public System.Guid[] Identifiers { get; }
        public string PropertyName { get; }
        public System.Type SagaType { get; }
    }
    public interface IAzureTableStorageSession
    {
        Microsoft.Azure.Cosmos.Table.TableBatchOperation Batch { get; }
        string PartitionKey { get; }
        Microsoft.Azure.Cosmos.Table.CloudTable Table { get; }
    }
    public interface IProvideCloudTableClient
    {
        Microsoft.Azure.Cosmos.Table.CloudTableClient Client { get; }
    }
    public interface IProvideCloudTableClientForSubscriptions
    {
        Microsoft.Azure.Cosmos.Table.CloudTableClient Client { get; }
    }
    public sealed class LogicalOutboxBehavior : NServiceBus.Pipeline.IBehavior, NServiceBus.Pipeline.IBehavior<NServiceBus.Pipeline.IIncomingLogicalMessageContext, NServiceBus.Pipeline.IIncomingLogicalMessageContext>
    {
        public System.Threading.Tasks.Task Invoke(NServiceBus.Pipeline.IIncomingLogicalMessageContext context, System.Func<NServiceBus.Pipeline.IIncomingLogicalMessageContext, System.Threading.Tasks.Task> next) { }
    }
    public class RetryNeededException : System.Exception
    {
        public RetryNeededException() { }
        public RetryNeededException(System.Exception innerException) { }
    }
    public static class SynchronizedStorageSessionExtensions
    {
        public static NServiceBus.Persistence.AzureTable.IAzureTableStorageSession AzureTablePersistenceSession(this NServiceBus.Persistence.SynchronizedStorageSession session) { }
    }
}
namespace NServiceBus.Testing
{
    public class TestableAzureTableStorageStorageSession : NServiceBus.Persistence.AzureTable.IAzureTableStorageSession, NServiceBus.Persistence.SynchronizedStorageSession
    {
        public TestableAzureTableStorageStorageSession(NServiceBus.TableEntityPartitionKey partitionKey) { }
        public Microsoft.Azure.Cosmos.Table.TableBatchOperation Batch { get; set; }
        public string PartitionKey { get; }
        public Microsoft.Azure.Cosmos.Table.CloudTable Table { get; set; }
    }
}